### ISSUE TITLE
Feature Request: Enable deploy in Remix dev mode

### ISSUE THREAD

**Issue by JQuezada0** [sst/sst#3985]
In SST V2 there was a field called [dev.deploy](https://v2.sst.dev/constructs/RemixSite#devdeploy) on the Remix construct (and others) that would allow you to run `sst dev` but still deploy the infrastructure.

Are there any thoughts on bringing something like that back? What I'm imagining is effectively a toggle between "fully deployed infra in dev mode with multiplexer running FE" and "fully deployed dev mode without multiplexer running FE".

**Comment by jayair** [comment]
You could use this for that case https://sst.dev/docs/component/experimental/dev-command

**Comment by SolomidHero** [comment]
@thdxr @jayair 

I meet same need, but maybe didn't understand well the approach discussed with `dev: false`.

**Basically, I need** to run site locally (as usually in `sst dev` functions). So far it is called "with multiplexer".
But I also want to share/publish such dev stage with public url I could use in other services.
This is the thing I see happens with `sst.aws.Function` resource. So I expect `sst.aws.Remix` to work like `sst.aws.Function` when `sst dev`.

With `dev: false` I get everything (lambdas, remix site, cdn, cloudflare, etc) deployed, and I see nothing runs locally.
So deployed lambdas of remix site are not working as stub servers anymore.
Thus, I believe that is not solution of my case.

How I should make such approach with sst only?

p.s. I could do `ngrok` tunneling to share url of my local site running, but I expect aws resources created for it as in `sst.aws.Function` case.

**Comment by jayair** [comment]
Yeah the only solution would be to have something like ngrok built into SST. We do have some thoughts around this.

**Comment by inconshreveable** [comment]
👋 ngrok founder here, if you're interested in baking something like ngrok in, we have SDKs like https://pkg.go.dev/golang.ngrok.com/ngrok and ngrok-js https://github.com/ngrok/ngrok-javascript. hope that's helpful!

**Comment by SolomidHero** [comment]
@jayair Can you please explain (for me and future people here) why it is not possible to use same approach to get permanent url as we get in `sst.aws.Function` dev?

**Comment by jayair** [comment]
In the case of Remix (and other frontends), `sst dev` starts the local dev mode of those frameworks. These are run in localhost.

The Function on the other hand is deployed and the requests are proxied to your local machine. So there's a real URL for it.
